src/gentropy/datasource/pan_ukbb_ld/ld.py

```diff
@@ -0,0 +1,214 @@
+            pan_ukbb_bm_path (str): Path to hail block matrix
+            ld_populations (list[str]): List of populations
+            ukbb_annotation_path (str): Path to pan-ukbb variant LD index with alleles flipped to match the order in OT variant annotation
+        Default values are set in PanUKBBConfig.
+        """
+        self.pan_ukbb_ht_path = pan_ukbb_ht_path
+        self.pan_ukbb_bm_path = pan_ukbb_bm_path
+        self.ld_populations = ld_populations
+        self.ukbb_annotation_output_path = ukbb_annotation_path
+
+    def align_ld_index_alleles(
+        self,
+        variant_annotation: DataFrame,
+        population: str,
+        hail_table_path: str = PanUKBBConfig.pan_ukbb_ht_path,
+        hail_table_output: str = PanUKBBConfig.ukbb_annotation_path,
+    ) -> None:
+        """Align Pan-UKBB variant LD index alleles with the Open Targets variant annotation.
+
+        Args:
+            variant_annotation (DataFrame): Open Targets variant annotation DataFrame
+            population (str): Population label
+            hail_table_path (str): Path to hail table with Pan-UKBB variant LD index
+            hail_table_output (str): Path to output the aligned Pan-UKBB variant LD index with alleles in the correct order
+        """
+        ht = hl.read_table(hail_table_path.format(POP=population))
+        ht = (
+            ht.to_spark()
+            .select(
+                "`locus.contig`",
+                "`locus.position`",
+                "`alleles`",
+                "`idx`",
+            )
+            .withColumns(
+                {
+                    "chromosome": f.split("`locus.contig`", "chr")[1],
+                    "position": f.col("`locus.position`"),
+                    "referenceAllele": f.element_at("`alleles`", 1),
+                    "alternateAllele": f.element_at("`alleles`", 2),
+                }
+            )
+            .drop("locus.contig", "locus.position", "alleles")
+            .dropDuplicates(
+                ["chromosome", "position", "referenceAllele", "alternateAllele"]
+            )
+        )
+        ht_va = (
+            ht.alias("ukbb")
+            .join(
+                variant_annotation.select(
+                    "chromosome",
+                    "position",
+                    f.col("referenceAllele").alias("va_ref"),
+                    f.col("alternateAllele").alias("va_alt"),
+                ).dropDuplicates(["chromosome", "position", "va_ref", "va_alt"]),
+                on=["chromosome", "position"],
+                how="left",
+            )
+            .filter(
+                (
+                    (f.col("referenceAllele") == f.col("va_ref"))
+                    & (f.col("alternateAllele") == f.col("va_alt"))
+                )
+                | (
+                    (f.col("referenceAllele") == f.col("va_alt"))
+                    & (f.col("alternateAllele") == f.col("va_ref"))
```

Comment:
If you are checking allele flipping, shouldn't the strandedness be checked as well? Eg. a `G` to `A` might be equivalent to a `T` to `C` change on the other strand.

Reply:
We don't have strand information in either dataset, so not able to check for this